src/Task1/SelectGenre.jsx

```diff
@@ -1,24 +1,29 @@
 import React, { useState } from "react";
 import "./selectGenre.css";
+import checkmark from "./../images/checkmark.svg";
 const SelectGenre = () => {
-  const [show, setShow] = useState("");
-  const [clip, setClippath] = useState("polygon(50% 75%, 0 25%, 100% 25%)");
+  const [isopen, setOpen] = useState(false);
+  const genre = ["Crime", "Documentary", "Horror", "Comedy"];
+  const [isChecked, setCheckmarks] = useState([false, false, false, false]);
+  const setCheck = (i) => {
+    setCheckmarks((prev) => {
+      let temp = [...prev];
+      temp[i] = !temp[i];
+      return temp;
+    });
+  };
   return (
     <div style={{ display: "flex", justifyContent: "center" }}>
       <div
-        id="list1"
-        className={"dropdown-check-list " + show}
-        tabindex="100"
+        className={`dropdown-check-list ${isopen ? "visible" : ""}`}
         style={{ width: "400px", backgroundColor: "#232323", color: "#FFFFFF" }}
       >
         <span
           onClick={() => {
-            if (show === "visible") {
-              setShow("");
-              setClippath("polygon(50% 75%, 0 25%, 100% 25%)");
+            if (isopen) {
+              setOpen(false);
             } else {
-              setShow("visible");
-              setClippath("polygon(50% 25%, 0 75%, 100% 75%)");
+              setOpen(true);
```

Comment:
Here again, the new state you are setting is dependent on the previous state. In this case, this dependency is a little bit hidden but if you rephrase your code in this logically identical way, the dependency will become more vivid:
```js
const newIsOpen = isOpen ? false : true // or just "const newIsOpen = !isOpen"
setOpen(newIsOpen)
```

Thus, if you remember one of my previous comments, when your new state depends on previous state, you should use callback argument to update state. So, in this case it would be just:
```js
setOpen((prevIsOpen) => !prevIsOpen)
```

Reply:
Yes, I fixed it

Reply:
Seems like it was not very clear. The last snippet was the goal. What you currently did is the 1st snippet, and it was just a simplification of your previous code, but still with issues. Simplification was to more clearly show you the issue.

Anyways, this is the goal:
```js
setOpen((prevIsOpen) => !prevIsOpen)
```

Reply:
I did